ocs/ocs_feed.py

```diff
@@ -226,30 +229,80 @@ def publish_message(self, message, timestamp=None):
             self.agent.publish(self.address, (message, self.encoded()))
 
     @staticmethod
-    def verify_message_data_type(message):
+    def verify_message_data_type(value):
         """Aggregated Feeds can only store certain types of data. Here we check
         that the type of all data contained in a message's 'data' dictionary are
         supported types.
 
         Args:
-            message (dict):
-                Data to be published (see Feed.publish_message for details).
+            value (list, float, int):
+                'data' dictionary value published (see Feed.publish_message for details).
 
         """
         valid_types = (float, int)
 
-        for k, v in message['data'].items():
-            # multi-sample check
-            if isinstance(v, list):
-                if not all(isinstance(x, valid_types) for x in v):
-                    type_set = set([type(x) for x in v])
-                    invalid_types = type_set.difference(valid_types)
-                    raise TypeError("message 'data' block contains invalid data" +
-                                    f"types: {invalid_types}")
+        # multi-sample check
+        if isinstance(value, list):
+            if not all(isinstance(x, valid_types) for x in value):
+                type_set = set([type(x) for x in value])
+                invalid_types = type_set.difference(valid_types)
+                raise TypeError("message 'data' block contains invalid data" +
+                                f"types: {invalid_types}")
 
-            # single sample check
-            else:
-                if not isinstance(v, valid_types):
-                    invalid_type = type(v)
-                    raise TypeError("message 'data' block contains invalid " +
-                                    f"data type: {invalid_type}")
+        # single sample check
+        else:
+            if not isinstance(value, valid_types):
+                invalid_type = type(value)
+                raise TypeError("message 'data' block contains invalid " +
+                                f"data type: {invalid_type}")
+
+    @staticmethod
+    def verify_data_field_string(field):
```

Comment:
The functionality here, and in the aggregator._enforce_field_name_rules is very similar.  Perhaps a single function that serves both purposes?  That would put the rules in one place, and there would only be one function to change / test / be aware of.

The converter is an implicit checker, afterall... if standardize(x) == x then x is already compliant.  (I know there's more to it than that, for the purposes of nice error text... but still, this could all be made to live in a single place.)

Reply:
I suppose. I think this came about from thinking of the Feed system first, with the thought that if everything is updated simultaneously (which is now the plan) it'll be impossible for invalid field names to get to the aggregator. This `Feed.verify_data_field_string()` method is used in the aggregator `_verify_provider_data()` method, which checks field names in the aggregator and marks them as needing to be modified before writing to disk.

A merged function would have to track the various reasons a field name was invalid to report errors in the instance of the Feed where we want the Agent to exit with a raised exception, vs in the aggregator where we don't want that behavior. 

I do think that even if this was changed we'd want to be testing it via test calls to the feed `Feed.publish_message()` and aggregator `Provider.save_to_block()`, so I'm not sure it actually reduces the number of tests run.

Admittedly, it was this separation that lead to missing several test cases that were caught by this PR, but now that we have it mostly working I'm somewhat resistant to rewriting it. I can do it if you feel strongly about it though. Do we anticipate changing these rules much in the future? (Such that it would be inconvenient to have the rule check and enforcement separate.)